src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.DataConnection.cs

```diff
@@ -118,7 +116,7 @@ public bool AllowRemove
                 if (CurrencyManager is not null)
                 {
                     // we only allow deletion on an IBindingList
-                    return (CurrencyManager.List is IBindingList) && CurrencyManager.AllowRemove && ((IBindingList)CurrencyManager.List).SupportsChangeNotification;
+                    return (CurrencyManager.List is IBindingList bindingList) && CurrencyManager.AllowRemove && bindingList.SupportsChangeNotification;
```

Comment:
```suggestion
                    return CurrencyManager is { AllowRemove: true, List: IBindingList { SupportsChangeNotification: true } };;
```
and you can remove the null check on `CurrencyManager`